benchexec/tools/template.py

```diff
@@ -22,17 +35,386 @@ class UnsupportedFeatureException(benchexec.BenchExecException):
     pass
 
 
-class BaseTool(object):
+class BaseTool2(object, metaclass=ABCMeta):
```

Comment:
I did this on purpose to make it easier to understand. When inheriting from `ABC`, it is not clear that this entails (there could be inherited methods, for example), whereas with the explicit metaclass it is clear that only the metaclass is changed.

Reply:
I'd argue it's easier to understand if there's less cognitive load. `ABC` should be just as well known as `ABCMeta`, so I don't think there's a risk of people fearing that there are any methods inherited.

Reply:
My thought was that most developers of tool-info modules are unfamiliar with all of `abc`, but using `metaclass` instead of inheritance indicates to them that there is less (nothing) they need to care and know about.